cpi-side/services/search-data.service.ts

```
@@ -1,50 +1,99 @@
-import { IClient, IContext } from '@pepperi-addons/cpi-node/build/cpi-side/events';
-import { filter, toApiQueryString } from '@pepperi-addons/pepperi-filters'
-import { GetValueOption, SearchDataConifuration } from 'shared';
+import { IClient } from '@pepperi-addons/cpi-node/build/cpi-side/events';
+import { toApiQueryString } from '@pepperi-addons/pepperi-filters';
+import { SearchDataConfiguration } from 'shared';
 import BaseCpiService from './base-cpi.service';
+import { AddonsDataSearchResult } from "@pepperi-addons/cpi-node/build/cpi-side/client-api";
+import { AddonData } from "@pepperi-addons/papi-sdk";
+
+interface CompositeConfiguration {
+    KeyFields: string[];
+    CompositeKeyDelimiter: string | undefined;
+}
 
 class SearchDataCpiService extends BaseCpiService {
 
-    private async getSearchData(configuration: SearchDataConifuration | undefined, context: any): Promise<any> {
-        let searchData: any;
+    private createCompositeDocumentKey(configuration: CompositeConfiguration): string {
+        const { KeyFields, CompositeKeyDelimiter } = configuration;
+        return KeyFields.join(CompositeKeyDelimiter);
+    }
 
-        if (configuration && configuration.Resource) {
-            let where: any = undefined;
+    private async getSearchData(configuration: SearchDataConfiguration, context: any): Promise<AddonsDataSearchResult | AddonData | null> {
+        if (!configuration || !configuration.Resource) {
+            return null;
+        }
+        const where = this.buildQueryString(configuration, context);
+        // Handle KeyType retrieval
+        const documentKey = this.getDocumentKey(configuration, context);
```

this is not the correct logic.. if the configuration is set to get the data by the key, and the key is empty you shouldn't search! but rather throw an exception or return undefined. either way there should be an console.error in here to point out that the configuration is broken

---

Yes, I think I am handling it in the same way. 
` const documentKey = this.getDocumentKey(configuration, context);
if(documentKey){
// then go for getBykey search
}
else we are going for where clause search.
`

let me know if I still need to make any changes here.

---

It was handled differently. On Ui, if the user changes the Search type by multiple and adds the Source query, we are resetting the KeyFields. [Here is the code](https://github.com/Pepperi-Addons/basic-logic-blocks/pull/14/files#diff-f287809f47b6c29ea689956372755a8c01b3ade7c126c5fadfab78dfe3aeb9d7R218)

So, at a time, it will only get one query, either By single key or by multiple resources by query.